zebROS_ws/src/behaviors/src/path_intake_actionlib_server.cpp

```diff
@@ -0,0 +1,165 @@
+    ros::NodeHandle nh_;
+    actionlib::SimpleActionServer<behaviors_actions::IntakeAction> as_;
+    std::string action_name_;
+    behaviors_actions::PathIntakeFeedback feedback_;
+    behaviors_actions::PathIntakeResult result_;
+
+  public:
+    PathIntakeAction(std::string name):
+      // boost bind is like a callback
+      as(nh), name, boost::bind(&PathIntakeAction::executeCB, this, _1), false),
+      action_name_(name)
+    {
+      as_.start();
+    }
+
+    ~PathIntakeAction(void){
+    }
+
+    void executeCB(const behavior_actions::PathIntakeGoalConstPtr &goal){
+      ros::Rate rate(1);
+      bool success = true;
+      bool preempted = false;
+      bool timed_out = false;
+      feedback_.percentage_done = 0; // not implemented yet
+
+      ROS_INFO_STREAM(action_name_.c.str() << ": Following path and intaking");
+
+      // call path generation server which returns path
+
+      nav_msgs::Path path_;
+
+      ros::ServiceClient game_piece_path_gen_client = nh.serviceClient<behavior_actions::GamePiecePickup>("game_piece_path_gen");
+      behaviors::GamePiecePickup game_piece_path_gen_srv;
+      if(game_piece_path_gen_client.call(game_piece_path_gen_srv)){
+        path_ = game_piece_path_server.response.path;
+      }
+
+      as_.publishFeedback(feedback_);
+
+      // call path following PathAction
+      if(!path_ac.waitForServer(ros::Duration(5))){
+					shutdownNode(ERROR,"Couldn't find path actionlib server");
+					return 1;
+			}
+      path_follower_server::PathGoal goal;
+      goal.path = path_;
+      path_ac.sendGoal(goal);
+      waitForActionlibServer(path_ac, 100, "running path"); // iterate??
+
+      as_.publishFeedback(feedback_);
+
+      // call intake IntakeAction
+      if(!intake_ac.waitForServer(ros::Duration(5))){
+					shutdownNode(ERROR,"Couldn't find intake actionlib server");
+					return 1;
+			}
+      behavior_actions::IntakeGoal goal;
+			intake_ac.sendGoal(goal);
+      waitForActionlibServer(intake_ac, 100, "running intake");
+
+      as_.publishFeedback(feedback_);
+
+      // retract intake by preempting (make sure this is how intake works)
+      intake_ac.cancel_goal();
+
+      if(as_.isPreemptRequested() || !ros::ok()){
+        preempted = true;
+        as_.setPreempted();
+      }
+
+      //log state of action and set result of action
+			behavior_actions::IntakeResult result;
+			result.timed_out = timed_out;
+			if(!success || timed_out)
+			{
+				ROS_WARN("%s: Error / Timed Out", action_name_.c_str());
+				result.success = false;
+				as_.setSucceeded(result);
+			}
+			else if(preempted)
+			{
+				ROS_WARN("%s: Preempted", action_name_.c_str());
+				result.success = false;
+				as_.setPreempted(result);
+			}
+			else //implies succeeded
+			{
+				ROS_WARN("%s: Succeeded", action_name_.c_str());
+				result.success = true;
+				as_.setSucceeded(result);
+			}
+
+    }
+
+
+}
+
+void waitForActionlibServer(T &action_client, double timeout, const std::string &activity)
```

Comment:
I think you might need `template <typename T>` like line 11, but changing it in line 11 may fix it here without having to change it